src/actions/router.js

```diff
@@ -1,14 +1,55 @@
 import parseUrl from 'grape-web/lib/parse-url'
 import * as router from 'react-router-redux'
 import find from 'lodash/collection/find'
+import {matchPath} from 'react-router-dom'
 
 import conf from '../conf'
 import * as types from '../constants/actionTypes'
 import {channelsSelector} from '../selectors'
 import {findLastUsedChannel} from './utils'
+import {channelRoute, pmRoute} from '../constants/routes'
+import {isChatUrl} from '../components/grapedown/utils'
 
 import {setChannel, openPm, openChannel, handleBadChannel} from './'
 
+const goToFromEmbedded = path => (dispatch) => {
+  const {pathname, hostname} = parseUrl(path)
+  const match = matchPath(pathname, {path: channelRoute}) ||
+    matchPath(pathname, {path: pmRoute})
+  // Open link in new tab, if it's outgoing link, or if link not leads to chat.
+  if (!match) {
+    window.open(path)
+    return
+  }
+  const {channelId, mateId, messageId} = match.params
+  const currentId = Number(channelId) || Number(mateId)
+  // Open link in the same window, because channelId of parsed link and opened chanelId similar
+  if (currentId === conf.channelId) {
+    if (!isChatUrl(path)) {
+      window.open(path)
+      return
+    }
+    // When link don't have messageId for current channel link, do nothing
+    // Examples: `@room`, `https://github.com/chat/channel/channelId`
+    if (!messageId) return
```

Comment:
"// When path doesn't contain a messageId, we don't have to do anything, because we are already in this channel
// Examples: `@room`, `https://github.com/chat/channel/channelId`
l".